### PR TITLE
[ENH] ncx2 dispatch to chi2 when nc=0

### LINKED ISSUE #5441: scipy.stats.ncx2 fails for nc=0
For `nc = 0` it should reduce to the chi-squared distribution, but `scipy.stats.ncx2.rvs(1, 0)` gives:

``` python
C:\Anaconda\lib\site-packages\scipy\stats\_distn_infrastructure.pyc in rvs(self, *args, **kwds)
    882         cond = logical_and(self._argcheck(*args), (scale >= 0))
    883         if not np.all(cond):
--> 884             raise ValueError("Domain error in arguments.")
    885 
    886         # self._size is total size of all output values

ValueError: Domain error in arguments.
```

Similarly the pdf, etc. also fail.

### PR DESCRIPTION
**Pull request by makbigc**
- close #5441
- one test added

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -3258,6 +3258,14 @@ def test_ncx2_tails_pdf():
     assert_(np.isneginf(logval).all())
 
 
+def test_ncx2_zero_nc():
+    # gh-5441
+    # ncx2 with nc=0 is identical to chi2
+    result = stats.ncx2.cdf(1, nc=0, df=10)
+    expected = stats.chi2.cdf(1, df=10)
+    assert_equal(result, expected)
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: please use something along the lines of `assert_allclose(result, expected, atol=1e-14)` for floating-point comparisons.

Comment (ev-br) [reply scipy/stats/tests/test_distributions.py]: Need also test pdf and logpdf. And I'd add at least a smoke test for an array_like `nc`, e.g. [4, 0].

Comment (ev-br) [reply scipy/stats/tests/test_distributions.py]: Please also add a test for `rvs`: recent numpies should be doing the right thing, however better check if our oldest supported numpy version is compatible.

**Review by ev-br — CHANGES_REQUESTED** [review]
Looks like a very good start! Test coverage should be expanded a bit though. However, if it accepts `nc=0`, all methods should work, so please test also `ppf` and `rvs`.

An alternative approach would be to do the dispatch on `nc==0` in the C code for ncx2 methods in special/cephes, but that's more work, and this approach is OK.

**Comment by ev-br** [comment]
PEP8 check is unhappy: 

```
40.16s$ pycodestyle scipy benchmarks/benchmarks
scipy/stats/tests/test_distributions.py:3277:38: E241 multiple spaces after ','
1       E241 multiple spaces after ','
The command "pycodestyle scipy benchmarks/benchmarks" exited with 1.
```

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -3258,6 +3258,26 @@ def test_ncx2_tails_pdf():
     assert_(np.isneginf(logval).all())
 
 
+@pytest.mark.parametrize('method, expected', [
+    ('cdf', np.array([1.72115630e-04, 2.74102033e-05])),
+    ('pdf', np.array([0.00078975, 0.00013013])),
+    ('logpdf', np.array([-7.14378973, -8.94700192]))
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: Please comment on where are these numbers from (imagine that ten years down the line, someone stares at these lines; your comment should help them check the numbers)

File: scipy/stats/tests/test_distributions.py [hunk 3]
```diff
@@ -3258,6 +3258,26 @@ def test_ncx2_tails_pdf():
...
+])
+def test_ncx2_zero_nc(method, expected):
+    # gh-5441
+    # ncx2 with nc=0 is identical to chi2
+    result = getattr(stats.ncx2, method)(1, nc=[0, 4], df=10)
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: Can just parameterize with ncx2.pdf, ncx2.cdf, ncx2.logpdf and get rid of the getattr?

File: scipy/stats/tests/test_distributions.py [hunk 4]
```diff
@@ -3258,6 +3258,26 @@ def test_ncx2_tails_pdf():
...
+    assert_allclose(result, expected, rtol=1e-3, atol=0)
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: Why this combinbation of tolerances? 
Assuming that the numbers come from chi2 distribution for nc=0 and ncx2 for nc != 0, the tolerances could/should be made way tighter, close 1e-15 or some such.

File: scipy/stats/tests/test_distributions.py [hunk 5]
```diff
@@ -3258,6 +3258,26 @@ def test_ncx2_tails_pdf():
...
+
+
+def test_ncx2_zero_nc_rvs():
+    # gh-5441
+    # ncx2 with nc=0 is identical to chi2
+    result = stats.ncx2.rvs(df=10, nc=[0,4], random_state=1)
+    expected = np.array([18.25726558,  8.24331569])
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: Again, please comment on the origin of this number. If it's chi2.rvs, maybe just write it our?

File: scipy/stats/tests/test_distributions.py [hunk 6]
```diff
@@ -3258,6 +3258,26 @@ def test_ncx2_tails_pdf():
...
+    # gh-5441
+    # ncx2 with nc=0 is identical to chi2
+    result = getattr(stats.ncx2, method)(1, nc=[0, 4], df=10)
+    assert_allclose(result, expected, rtol=1e-3, atol=0)
+
+
+def test_ncx2_zero_nc_rvs():
+    # gh-5441
+    # ncx2 with nc=0 is identical to chi2
+    result = stats.ncx2.rvs(df=10, nc=[0,4], random_state=1)
```
Comment (ev-br) [inline scipy/stats/tests/test_distributions.py]: spaces after commas, please: `[0, 4]`

File: scipy/stats/tests/test_distributions.py (hunk 3 above)
Comment (makbigc) [reply scipy/stats/tests/test_distributions.py]: The parameter of the pytest decorator is regarded as `string` which doesn't have `__call__`.
Explicitly, if the parameters is set as follows:
```
@pytest.mark.parametrize('method, expected', [
    ('ncx2.cdf', np.array([2.497951336e-09, 3.437288941e-10])),
    ('ncx2.pdf', np.array([1.238579980e-07, 1.710041145e-08])),
    ('ncx2.logpdf', np.array([-15.90413011, -17.88416331]))
])
```
`method(0.1, nc=[0, 4], df=10)` would run into `TypeError`.

**Comment by makbigc** [comment]
@ev-br Do you have anything else to implement? Please tell me.

**Comment by ev-br** [comment]
`ppf` could also dispatch to `chi2.ppf`, however the present version seems to agree to `chi2.ppf` at a level of 1e-11, so it could be done in a follow-up PR.

Looks good now, merging. Thanks @makbigc

**Pull request merged**